userComponent/src/main/kotlin/com/gravatar/app/usercomponent/data/RealProfileRepository.kt

```diff
@@ -0,0 +1,69 @@
+package com.gravatar.app.usercomponent.data
+
+import com.gravatar.app.usercomponent.domain.repository.ProfileRepository
+import com.gravatar.restapi.models.Profile
+import com.gravatar.restapi.models.UpdateProfileRequest
+import com.gravatar.services.ProfileService
+import kotlinx.coroutines.flow.firstOrNull
+
+internal class RealProfileRepository(
+    private val profileService: ProfileService,
+    private val tokenStorage: AuthTokenStorage,
+) : ProfileRepository {
+
+    // Temporary in-memory storage for the profile
+    private var profile: Profile? = null
+
+    override suspend fun refreshUserProfile(): Result<Unit> {
+        return fetchProfile().fold(
+            onSuccess = {
+                Result.success(Unit)
+            },
+            onFailure = { error ->
+                Result.failure(error)
+            }
+        )
+    }
+
+    override suspend fun get(): Result<Profile> {
```

Comment:
This might be changed to `Flow<Profile`, but we aren't really using it like that at this, so I've kept it like so.